src/coreclr/src/jit/lower.cpp

```diff
@@ -296,8 +296,28 @@ GenTree* Lowering::LowerNode(GenTree* node)
         }
 
         case GT_LCL_VAR:
-            WidenSIMD12IfNecessary(node->AsLclVarCommon());
+        {
+            GenTreeLclVar* lclNode = node->AsLclVar();
+            WidenSIMD12IfNecessary(lclNode);
+            LclVarDsc* varDsc = comp->lvaGetDesc(lclNode);
+
+            // The consumer of this node must check compatibility of the fields.
+            // This merely checks whether it is possible for this to be a multireg node.
+            if (lclNode->IsMultiRegLclVar())
+            {
+                if (!varDsc->lvPromoted ||
+                    (comp->lvaGetPromotionType(varDsc) != Compiler::PROMOTION_TYPE_INDEPENDENT) ||
+                    (varDsc->lvFieldCnt > MAX_MULTIREG_COUNT))
+                {
+                    lclNode->ClearMultiReg();
```

Comment:
Could we check `(varDsc->lvFieldCnt > MAX_MULTIREG_COUNT)` during importation and avoid setting `MultiReg` for such nodes?

Reply:
I'd like to see us make these decisions in `Lowering`, since eventually we'd like to be able to promote register-passed structs with more than `MAX_MULTIREG_COUNT` fields (i.e. multiple fields packed into a single register).